src/ResourceManagement/Sql/CheckNameAvailabilityResultImpl.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+namespace Microsoft.Azure.Management.Sql.Fluent
+{
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using Microsoft.Azure.Management.Sql.Fluent.Models;
+
+    /// <summary>
+    /// Implementation for CheckNameAvailabilityResult.
+    /// </summary>
+///GENTHASH:Y29tLm1pY3Jvc29mdC5henVyZS5tYW5hZ2VtZW50LnNxbC5pbXBsZW1lbnRhdGlvbi5DaGVja05hbWVBdmFpbGFiaWxpdHlSZXN1bHRJbXBs
+    internal partial class CheckNameAvailabilityResultImpl  :
+        Wrapper<Models.CheckNameAvailabilityResponseInner>,
+        ICheckNameAvailabilityResult
+    {
+
+        /// <summary>
+        /// Creates an instance of the check name availability result object.
+        /// </summary>
+        /// <param name="inner">The inner object.</param>
+        ///GENMHASH:6E11439BC0EE92936D1373B518E18596:0FD5BCAE10307382B777A749B232A151
+        internal CheckNameAvailabilityResultImpl(CheckNameAvailabilityResponseInner inner) : base(inner)
+        {
+        }
+
+        ///GENMHASH:ECE9AA3B22E6D72ED037B235766E650D:853A4A58250437949396E2FC3A58AA01
+        public bool IsAvailable()
+        {
+            return this.Inner.Available.GetValueOrDefault(false);
+        }
```

Comment:
I do not think this pattern is good. If the inner type is nullable then our wrapped type also should be nullable. It is not good to return default value - in this case false, when server did not have this field in the response body. IMHO there is a difference between null and false value. And this logic is present all over SQL. If this is a new pattern I'd ask you to change it in this release.

Reply:
The return must be a boolean in this case, and "null" should be treated as "false". This API checks if the server name is available (SQL names are FQDN across the whole cloud).

Reply:
FYI we actually could do worst; check out Traffic Manager  CheckProfileDnsNameAvailabilityResult()

Reply:
@milismsft  Seems like another issue for github :) Could you add it? I'm done with the PR review. If you addressed all the comments and know what to do - you can merge :).

Reply:
Returning "false" for nullable bool or "0" for nullable integer, long or double is not new. We need to stay consistent with the Java implementation; whatever we return in Java we should return as default here as well. For instance if the return in Java type is "Integer" which can be "null", we return 0, not "null".

Reply:
I should add that for string, UUID/Guid or DateTime which are not primary types, we should always return the nullable type (in Java we return null too). In SQL I do this for the newer stuff but for the old stuff I could not change them because of risk of introducing a breaking change.